### PR TITLE
feat: add tracking log for completion events

### PR DESCRIPTION
**Pull request by Ian2012**
**Description:** This PR emits a tracking log for every completion event (created/updated). 

**Related issue:** https://github.com/openedx/openedx-aspects/issues/88

**Installation instructions:** List any non-trivial installation 
instructions.

**Testing instructions:** 

- Create a problem in an empty unit.
- Go to the LMS and finish the problem.
- Verify a tracking log named: `edx.completion.block_completion.changed` is emitted with the proper information

**Reviewers:**
- [ ] tag reviewer 
- [ ] tag reviewer 

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Ian2012!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by bmtcril** [comment]
Pinged the last couple of people who seem to have done work in this repo, lmk if there are better reviewers. 😄

File: completion/tests/test_models.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def setUp(self):
         self.set_up_completion()
 
     def test_changed_value(self):
-        with self.assertNumQueries(6):  # Get, update, 2 * savepoints, 2 * exists checks
+        with self.assertNumQueries(7):  # 2 * Get, update, 2 * savepoints, 2 * exists checks
```
Comment (bmtcril) [inline completion/tests/test_models.py:44]: These are a little concerning, do you know where the extra queries are coming from?

Comment (Ian2012) [reply completion/tests/test_models.py:44]: There is one additional for the `obj.emit_tracking_log()` method.

**Comment by Ian2012** [comment]
Hey @mphilbrick211! Is there any reviewer for this PR?

This tracking log is really important for aspects

**Comment by mphilbrick211** [comment]
Hi @openedx/masters-devs-cosmonauts! Would someone please be able to get this reviewed/merged for us? Thanks!

**Comment by Zacharis278** [comment]
@mphilbrick211 adding a ticket to our backlog to handle the merge and release for this

File: completion/models.py [hunk 2]
```diff
@@ -321,3 +330,18 @@ class Meta:
...
+        Emit a tracking log when a block completion is created or updated.
+        """
+        tracker.emit(
+            BLOCK_COMPLETION_CHANGED_EVENT_TYPE,
+            {
+                'user_id': self.user.id,
+                'course_id': str(self.context_key),
+                'block_id': str(self.block_key),
+                'block_type': self.block_type,
+                'completion': self.completion,
```
Comment (Ian2012) [inline completion/models.py:345]: Author concern: add the modfiied/created timestamp for backfill puposes

**Comment by Ian2012** [comment]
@Zacharis278 Is there anything we need to do before merging this PR?

**Comment by mphilbrick211** [comment]
Hi @Zacharis278! Is there an ETA on this? I know you mentioned it was being added to your backlog.

Checking in because it's directly blocking [this PR](https://github.com/openedx/event-routing-backends/pull/339) which is important to the Aspects work. Please let us know if there's any updates. Thanks!

**Comment by Ian2012** [comment]
cc @UsamaSadiq @zubairshakoorarbisoft Can you take a look at this PR?

This is really important event for the Aspects project

**Comment by bmtcril** [comment]
Looks good to me!

**Review by UsamaSadiq — APPROVED** [review]
Looks good to me.

**Comment by openedx-webhooks** [comment]
@Ian2012 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/completion&entry.867055334=https://github.com/openedx/completion/pull/245&entry.1484655318=https://github.com/Ian2012&entry.752974735=2023-08-22+18:12&entry.1917517419=2023-09-21+18:02&entry.2133058324=Yes) so we can improve your experience in the future.

**Comment by bmtcril** [comment]
Hi @UsamaSadiq thanks for merging, can someone roll a release as well? Looks like there hasn't been one in a while.

**Pull request merged**